scrapy/linkextractors/__init__.py

```diff
@@ -101,7 +101,7 @@ def _process_links(self, links):
         links = [x for x in links if self._link_allowed(x)]
         if self.canonicalize:
             for link in links:
-                link.url = canonicalize_url(urlparse(link.url))
+                link.url = canonicalize_url(link.url)
```

Comment:
@redapple do you recall why did you use urlparse here? Is it just a left-over from refactoring, or is it something else?